core/test/variable_operations_test.cpp

```diff
@@ -370,6 +370,16 @@ TEST(Variable, operator_allowed_types) {
   /* Can not operate on lower precision from higher precision */
   EXPECT_ANY_THROW(i32 += i64);
   EXPECT_ANY_THROW(f += d);
+
+  /* Expcet promotion to double if one parameter is double */
+  {
+    const auto r = f + d;
+    EXPECT_EQ(DType::Double, r.dtype());
```

Comment:
I recommend using the variable template `dtype<double>` instead of the enum values.